### PR TITLE
Be/keyword arguments

### PR DESCRIPTION
**Pull request by beccaelenzil**
Keyword arguments. Almost a direct move from textbook curriculum.

### CODE REVIEW

**Review by beccaelenzil — COMMENTED** [review]
I'm not sure how this difference in the conditional lessons was introduced into this PR

File: 02-methods/10-keyword-arguments.md [hunk 1]
```diff
@@ -0,0 +1,299 @@
+# Keyword and Optional Arguments
+
+## Learning Goals
+- Differentiate between _positional_ and _keyword_ arguments
```
Comment (CheezItMan) [inline 02-methods/10-keyword-arguments.md:4]: Minor formatting just keeping the pattern Simon had introduced.

```suggestion

By the end of this lesson, we will be able to...

- Differentiate between _positional_ and _keyword_ arguments
```

File: 02-methods/10-keyword-arguments.md [hunk 2]
```diff
@@ -0,0 +1,299 @@
...
+##### !end-placeholder
+
+##### !answer
+
+/coffee_price\s?(type,\s?size\s?)/
+
+##### !end-answer
+
+
+
```
Comment (CheezItMan) [inline 02-methods/10-keyword-arguments.md:96]: I suggest adding an explanation or hint just to make it clear to students.

**Review by CheezItMan — APPROVED** [review]
A few comments, but this is good to go

File: 02-methods/10-keyword-arguments.md [hunk 3]
```diff
@@ -0,0 +1,299 @@
...
+```
+
+Notice that not all the arguments in the method above are optional. Optional Arguments can co-exist with Positional Arguments (and often do).
+
+Positional arguments must come first, both in the definition and when the method is called.
+
+Optional arguments can help make method invocations easier to read and write, and you'll see them frequently in Ruby, particularly when working with gems or other peoples' code. However, the following issues:
+
+1. The caller must remember the order of the arguments
+1. When reading the call, there's no way to tell which argument is which
```
Comment (kaidamasaki) [inline 02-methods/10-keyword-arguments.md:211]:
```suggestion
1. The caller must remember the order of the arguments
2. When reading the call, there's no way to tell which argument is which
```

File: 02-methods/10-keyword-arguments.md [hunk 4]
```diff
@@ -0,0 +1,299 @@
...
+# Price for a tall iced drip
+coffee_price(:drip, :tall, extra_shots: 0, cold: true)
+
+# Price for a grande cappuccino
+coffee_price(:cappuccino, :grande, extra_shots: 0, cold: false)
+```
+
+There are a few things to note here:
+- When a method is invoked, any keyword arguments are passed by name (a.k.a. _keyword_)
+- Keyword arguments can be passed in any order. (ie. `coffee_price(:drip, :tall, cold: true, extra_shots:2`))
```
Comment (kaidamasaki) [inline 02-methods/10-keyword-arguments.md:238]:
```suggestion
- Keyword arguments can be passed in any order. (ie. `coffee_price(:drip, :tall, cold: true, extra_shots:2)`)
```

File: 02-methods/10-keyword-arguments.md [hunk 5]
```diff
@@ -0,0 +1,299 @@
...
+coffee_price(:latte, :venti, extra_shots: 3, cold: false)
+
+# Price for a tall iced drip
+coffee_price(:drip, :tall, cold: true)
+
+# Price for a grande cappuccino
+coffee_price(:cappuccino, :grande)
+```
+
+In each of these iterations, the only things we've changed are the parameters in the method signature, the body of the method remains the same. The end result should resemble [this code](source/CoffeePrice.rb).
```
Comment (kaidamasaki) [inline 02-methods/10-keyword-arguments.md:265]: It looks like this PR doesn't include this file.

Comment (kaidamasaki) [reply 02-methods/10-keyword-arguments.md:265]:
```suggestion
In each of these iterations, the only things we've changed are the parameters in the method signature, the body of the method remains the same. 

<details>
  <summary>Expand this to see the end result</summary>
  
  <pre lang="ruby">
def coffee_price(type, size, extra_shots: 0, cold: false)
  # Find the base price for this drink
  case type
  when :drip
    price = 1.5
  when :latte
    price = 3.7
  when :cappuccino
    price = 3.2
  else
    puts "Invalid coffee type: #{type}"
    return
  end
  
  # Modify for size
  case size
  when :tall
    # No change
  when :grande
    price *= 1.3
  when :venti
    price *= 1.6
  else
    puts "Invalid size: #{size}"
    return
  end
  # Charge for extra shots
  price += extra_shots * 0.5
  # Charge for a cold drink
  if cold
    price += 1
  end

  return price
end
  </pre>
</details>
```

File: 02-methods/10-keyword-arguments.md [hunk 6]
```diff
@@ -0,0 +1,299 @@
...
+| **Positional** | coffee_price(type, size, extra_shots, cold) | coffee_price(type, size, extra_shots = 0, cold = false) |
+| **Keyword**    | coffee_price(type:, size:, extra_shots:, cold:) | coffee_price(type, size, extra_shots: 0, cold: false) |
+
+- **Positional arguments** are the type of method arguments we've seen so far
+- **Keyword arguments** 
+  - are given by name
+  - can make calling a method easier when there are several arguments
+- **Optional arguments** 
+  - default to a specified value if not given by the caller
+  - both Positional and Keyword Arguments can be made Optional
```
Comment (kaidamasaki) [inline 02-methods/10-keyword-arguments.md:293]:
```suggestion
  - both _Positional Arguments_ and _Keyword Arguments_ can be made _Optional_
```

File: 02-methods/10-keyword-arguments.md [hunk 7]
```diff
@@ -0,0 +1,299 @@
...
+
+- Keyword Arguments and Optional Arguments can co-exist with Positional Arguments
```
Comment (kaidamasaki) [inline 02-methods/10-keyword-arguments.md:295]:
```suggestion
- _Keyword Arguments_ and _Optional Arguments_ can co-exist with _Positional Arguments_
```

File: 02-methods/10-keyword-arguments.md (hunk 5 above)
Comment (beccaelenzil) [reply 02-methods/10-keyword-arguments.md]: added code in details tag

**Review by CheezItMan — APPROVED** [review]
:shipit:

**Pull request merged**